### PR TITLE
ci(github): add CD workflow for master branch

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Implementing an initial CD workflow for the master branch

### Potential Breaking Changes

Deployment might be broken after merging this, but I'll take care to adjust if it is the case.

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: CD
+
+on:
+    push:
+        branches:
+            - master
```
Comment (GermainBergeron) [inline .github/workflows/cd.yml:6]: Should we have something here or a bit down below that skips if it was a version bump?

File: .github/actions/deploy/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        description: The AWS S3 bucket in which to deploy the website.
+runs:
+    using: composite
+    steps:
+        - name: Configure AWS Credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+              role-to-assume: ${{ inputs.AWS_ROLE }}
+              aws-region: ${{ inputs.AWS_REGION }}
+        - name: Push distribution code to s3 demo folder
```
Comment (aiheon) [inline .github/actions/deploy/action.yml:18]: This make me think that we should maybe add the `clean up demo` workflow in Plasma also? 🤔

Comment (gdostie) [reply .github/actions/deploy/action.yml:18]: Good idea, I will create a separate task for it and do it in a different PR

**Review by aiheon — APPROVED** [review]
I like it 🎉

**Pull request merged**